pkg/skaffold/ui/ui.go

```diff
@@ -0,0 +1,31 @@
+/*
+Copyright 2020 The Skaffold Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package ui
+
+import (
+	"io"
+	"io/ioutil"
+)
+
+var (
+	out io.Writer = ioutil.Discard
+)
+
+// SetOutput sets the destination for messages from the ui package
+func SetOutput(w io.Writer) {
```

Comment:
How this ui package going to be used? Would there be a way to not get progress bars and raw skaffold output?
IDEs like VScode and intellij use skaffold stream as is. This might not render well on the IDE side untill they switch to constructing UI based on event apis.

```
ui.NewProgressGroup()
topLevel := ui.AddNewSpinner("", "Building...", nil)

skaffold.Build()

topLevel.Increment()
ui.Wait()
```

Reply:
Under the changes that Brian suggested, a ProgressManager would be set at the beginning of the skaffold execution. By default this would be a null ProgressManager, and so added function calls would do nothing. If using the implementation that I'm adding here, then the function calls would spawn the progress indicators that I've made